src/zts.ts

```diff
@@ -28,6 +28,21 @@ const samples = {
   [ts.SyntaxKind.UndefinedKeyword]: undefined,
 } satisfies Partial<Record<ts.KeywordTypeSyntaxKind, unknown>>;
 
+const nodePath = {
+  name: path([
+    "name" satisfies keyof ts.TypeElement,
+    "text" satisfies keyof Exclude<
+      NonNullable<ts.TypeElement["name"]>,
+      ts.ComputedPropertyName
+    >,
+  ]),
+  type: path([
+    "type" satisfies keyof ts.PropertySignature,
+    "kind" satisfies keyof NonNullable<ts.PropertySignature["type"]>,
```

Comment:
what about same kind but different essence?
like objects, for example.
maybe should limit this to primitives only?